test/dynamo/test_export.py

```diff
@@ -3245,18 +3245,17 @@ def forward(self, pred, x):
         cos = arg1.cos();  arg1 = None
         return pytree.tree_unflatten([cos], self._out_spec)
 """
-        true_guard_code = ["cast_symbool_to_symint_guardless(L['pred']) == 1"]
+        true_guard_code = ["convert_symbool_to_int_with_hint(L['pred']) == 1"]
         false_guard_code = [
-            "Ne(cast_symbool_to_symint_guardless(L['pred']), 1)",
-            "-9223372036854775808 <= cast_symbool_to_symint_guardless(L['pred'])",
+            "Ne(convert_symbool_to_int_with_hint(L['pred']), 1)",
+            "-9223372036854775808 <= convert_symbool_to_int_with_hint(L['pred'])",
```

Comment:
This doesn't seem right? You are returning a guard and now the guard code is no longer triggering specialization on the outer make_fx call?

Reply:
yes, you're right. My current proposal is to propagate the guard after they've been generated and don't install guard when we run it second time because the guard has been installed. 

But as I'm writing, I find that this approach doesn't correctly handle a new make_fx session
```python
gm = make_fx(compiled_f)(x) # gm.shape_env.guards has guards installed
gm2 = make_fx(compiled_f)(x) # gm2.shape_env.guards don't have guard because guard checking didn't install guards.
```

We probably should make the guard code trigger specialization. If it's the same shape_env, doing it twice (once in SHAPE_ENV guard and once in guard checking) should be OK...

Reply:
yes exactly